crates/krator/src/runtime.rs

```diff
@@ -167,6 +172,13 @@ impl<O: Operator> OperatorRuntime<O> {
                             manifest.namespace()
                         );
                         reflector_deleted.notify();
+                        match manifest_tx.broadcast(manifest) {
+                            Ok(()) => (),
+                            Err(e) => {
+                                warn!("Unable to broadcast manifest update: {:?}", e);
```

Comment:
Nit: I think that in a deleted case, this might be better classified as an error as it is kinda important that it notifies it is going to be gone. But I could be wrong here

Reply:
I think this error would only happen if all `Receivers` are dropped. That would mean that `run_to_completion` has completed or panicked, so I don't think we can really do anything. I can bump the log level to `ERROR` if you wish.

Reply:
```
 it has to be possible for the watched objects to persist when the Krustlet is shut down, and in case it is deleted in the meantime, we need that notification.
```
I'm not sure I follow what you are saying here. If Krator shuts down (SIGINT), it actually doesnt "evict" any objects. This is functionality implemented in Krustlet. Right now Krator just becomes unschedulable essentially. It actually wont shut down. I'm not sure what the desired behavior here would be. As far as getting a "notification" that an object has been deleted (I'm assuming when Krator starts up again?), the deletion timestamp should be set so I believe this will work fine. 

Regarding finalizers:

1. If an object which Krator has a finalizer registered for is deleted while Krator is shut down, I think the object would get stuck, but this is expected behavior with finalizers (at least I experience it pretty often). 
2. If Krator is operating an object with Krator finalizers, I think this should all work, just do the finalizers in the `Deleted` state and then exit. The only thing we might need here is a way to remove the finalizer once it has run. 
3. If Krator is operating an object with external finalizers, there will probably be an error logged when it tries to delete the object if those have not completed. I'm not sure what is responsible for retrying that delete or if k8s will delete the object once the finalizers have run.

Reply:
My understanding of finalizers is that an object will not be fully deleted from Kubernetes as long as at least one finalizer is attached to it. Basically if Krator adds a finalizer to an object it is saying "do not under any circumstances delete this before I remove my finalizer, I will be back to do so!"
Without a finalizer it is my understanding that when deleting an object a deletion timestamp will be set, but the object could vanish before Krator comes back (crashed over the weekend, restarted monday, gone for a few days) - this scenario could be solved by using a finalizer.

You are right in that I can simply set the finalizer in my code and remove it in the deleted state, this is not necessarily something which needs to be part of Krator - I honestly hadn't even thought of that :)

Reply:
I see, it does sound like a finalizer would be a good way to ensure that Krator is able to do that cleanup. 

FWIW, my suggestion of setting / removing the finalizer is how I think they are intended to be used (each controller detects the deletion timestamp, runs its finalizer, and then patches the Pod to remove the finalizer.